src/models/Visitors.php

```diff
@@ -0,0 +1,344 @@
+<?php
+namespace DevGroup\Analytics\models;
+
+use Yii;
+use app;
+use yii\web\Session;
+use yii\web\Request;
+
+
+use DevGroup\Analytics\models\VisitedPage;
+use DevGroup\Analytics\models\Visitor;
+use DevGroup\Analytics\models\Cities;
+use DevGroup\Analytics\models\Countries;
+
+use DevGroup\Analytics\models\VisitorSession;
+
+class Visitors extends Session
+{
+
+    const KEY_COOKIE = 'visitor';
+
+    const VISITOR_NAME_KEY = 'visitor_id';
+
+    const SESSION_NAME_KEY = 'session_id';
+
+    const USER_IF_AUTO = 'user_status';
+
+
+    public $_statusAddLastVisitor = true;
+
+    public $_statusAddLastSession = true;
+
+    public $_statusAddUser = true;
+
+
+    /**
+     * @return int|mixed|null|string
+     */
+    public function getRefererId()
+    {
+        $referer = $this->getReferrer();
+
+        if ($referer !== null) {
+            return $this->getPageId($referer);
+        } else {
+            return null;
+        }
+    }
+
+    public function addLastVisitor($visitorId)
+    {
+        if ($this->_statusAddLastVisitor && $this->get($this::SESSION_NAME_KEY, false)) {
+            VisitorSession::updateAll([
+                'last_action_at' => date('Y-m-d H:i:s'),
+                'last_visited_page_id' => $this->getPageId(Yii::$app->request->url)
+            ], ['=', 'id', $visitorId]);
+        }
+    }
+
+    public function addLastSession($sessionId)
+    {
+        if ($this->_statusAddLastSession && $this->get($this::VISITOR_NAME_KEY, false)) {
+            // incorrect
+            Visitor::updateAll([
+                'last_activity_at' => date('Y-m-d H:i:s'),
+                'last_activity_visited_page_id' => $this->getPageId(Yii::$app->request->url)
+            ], ['=', 'id', $sessionId]);
+        }
+    }
+
+
+    /**
+     * @return int
+     */
+    private function sessionLifeTime()
+    {
+        return time() + 30;
+    }
+
+    public function run()
+    {
+
+        list($visitorId, $sessionId) = $this->init();
```

Comment:
`init` is yii's function that is called internally during object creation.
YOU SHOULD NEVER CALL IT
Also, list is not good as of the times of php4. What will be if someone redefines it? Why not to use variables of $this ?